components/carousel/carouselItem.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class CarouselItem extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      imageSrc: null,
+      hasLoaded: false,
+    };
+  }
+
+  componentDidMount() {
+    this.setupImage(this.props.load, this.props.src);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.load && !this.state.hasLoaded) {
+      this.setupImage(nextProps.load, nextProps.src);
+    }
+  }
+
+  setupImage(load, src) {
+    if (!load || !src) {
+      return;
+    }
+    this.setState(() => ({ hasLoaded: true, imageSrc: src }));
+  }
+
+  render() {
+    const { imageSrc, hasLoaded } = this.state;
+    const imageStyles = {};
+
+    if (imageSrc && hasLoaded) {
+      imageStyles.backgroundImage = `url('${imageSrc}')`;
```

Comment:
Looks like you don't need to keep track both of `imageSrc` and `hasLoaded` using `props.src` and `state.hasLoaded` is enough.